src/passes/OptimizeCasts.cpp

```diff
@@ -72,13 +72,18 @@
 // sense in new pass anyhow, and things should be simpler overall to keep such
 // casts all in one pass, here.
 //
-// TODO: Move casts earlier in a basic block as well, at least in traps-never-
-//       happen mode where we can assume they never fail.
-// TODO: Look past individual basic blocks?
-// TODO: Look at LocalSet as well and not just Get. That would add some overlap
-//       with the other passes mentioned above, but once we do things like
-//       moving casts earlier as in the other TODO, we'd be doing uniquely
-//       useful things with LocalSet here.
+// TODO: 1. Move casts earlier in a basic block as well, at least in
+//          traps-never-happen mode where we can assume they never fail, and
+//          perhaps in other situations too.
+// TODO: 2. Look past individual basic blocks? This may be worth considering
+//          given the pattern of a cast appearing in an if condition that is
+//          then used in an if arm, for example, where simple dominance shows
+//          the cast can be reused.
+// TODO: 3. Look at LocalSet as well and not just Get. That would add some
+//          overlap with the other passes mentioned above, but once we do the
+//          first two TODOs above then we'd be adding some novel things here.
```

Comment:
Can you explain more about what the novel parts would be and what corresponding improvements could be made to the other passes? This will help me understand more and will also inform our work this summer 🎉

Reply:
Sure, I added more details now.

Note that the 3rd TODO is a little speculative and needs more thought, as I added more on now. The first two definitely look worthwhile to me, but I'm not sure of the 3rd - let's reconsider that one later.